src/workerd/api/streams/standard.c++

```diff
@@ -2467,12 +2482,14 @@ kj::Maybe<jsg::Promise<ReadResult>> ReadableStreamJsController::read(
       // The ReadableStreamDefaultController does not support ByobOptions.
       // It should never happen, but let's make sure.
       KJ_ASSERT(maybeByobOptions == kj::none);
-      ReadPendingScope readPendingScope(js, *this);
-      return consumer->read(js);
+      return readPendingScope(js, *this, [&]() mutable {
```

Comment:
maybe a better name is possible? how about `ensureControllerAlive` or something like that?

Reply:
`runDeferringCloseAndErrorStateChanges()`? :)